SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h

```diff
@@ -52,8 +52,11 @@ class SPATIALGDK_API InterestFactory
 	// The components servers need to see on entities they have authority over that they don't already see through authority.
 	void AddServerSelfInterest(Interest& OutInterest) const;
 
-	void GetActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries, bool bRecurseChildren) const;
-	TArray<Query> GetUserDefinedQueries(const QueryConstraint& LevelConstraints) const;
+	TArray<Query> GetUserDefinedQueries(const AActor* InActor, const QueryConstraint LevelConstraint) const;
+	TMap<float, TArray<QueryConstraint>> GetUserDefinedFrequencyToConstraintsMap(const AActor* InActor) const;
```

Comment:
Just some comments that you might think are nice. Definitely not necessary.
- Might be nice to create a custom type of TMap<float, TArray<QueryConstraint>>
- There other small thing Some function return the frequency-constraint maps, some take a ref to a frequency-constraint map. Might be nice to use one method. Again not necessary.

Reply:
the one function that takes a map to populate is passed a reference to a map created in the function that returns the final result which calls the function that populates multiple times for different cases, so I think this makes sense? Happy to discuss further if I have misunderstood something

The frequency to array of constraints is a good idea

Reply:
Ahh yes no that makes a lot of sense. I didn't think about that, only thought from the perspective of the header. Nice.